### PR TITLE
Add client type validation

### PR DESCRIPTION
**Pull request by smolchanovsky**
Clients can be created using interface types, not classes. Thus, it is necessary to check that the generic parameter is an interface.

### CODE REVIEW

**Comment by Kingmidas74** [comment]
Some description is required

File: tests/NClient/NClient.Tests/ClientTests/OptionalParamClientTest.cs [hunk 1]
```diff
@@ -11,6 +11,17 @@ namespace NClient.Tests.ClientTests
     [Parallelizable]
     public class OptionalParamClientTest
     {
+        [Test, Order(0)]
+        public void OptionalParamClient_Build_NotThrow()
+        {
+            const string anyHost = "http://localhost:5000";
```
Comment (Kingmidas74) [inline tests/NClient/NClient.Tests/ClientTests/OptionalParamClientTest.cs:17]: I think we should use Fixture or smth fot generate some general values..

Comment (Kingmidas74) [reply tests/NClient/NClient.Tests/ClientTests/OptionalParamClientTest.cs:17]: Maybe not right here and now, just fyi

Comment (smolchanovsky) [reply tests/NClient/NClient.Tests/ClientTests/OptionalParamClientTest.cs:17]: I will use them. I just forgot that Fixture is already used in tests for deconstructors.

Comment (smolchanovsky) [reply tests/NClient/NClient.Tests/ClientTests/OptionalParamClientTest.cs]: Done

**Review by Kingmidas74 — APPROVED** [review]

**Pull request merged**